File: test/extensions/language/default_values/python/StructureDefaultValuesTest.py
import DefaultValues


class StructureDefaultValuesTest(DefaultValues.TestCase):
    def testDefaultBoolValue(self):
        structureDefaultValues = self.api.StructureDefaultValues()
        self.assertEqual(True, structureDefaultValues.bool_value)

    def testDefaultBit4Value(self):
        structureDefaultValues = self.api.StructureDefaultValues()
        self.assertEqual(0x0F, structureDefaultValues.bit4_value)

    def testDefaultInt16Value(self):
        structureDefaultValues = self.api.StructureDefaultValues()
        self.assertEqual(0x0BEE, structureDefaultValues.int16_value)

    def testDefaultFloat16Value(self):
        structureDefaultValues = self.api.StructureDefaultValues()
        self.assertEqual(1.23, structureDefaultValues.float16_value)

    def testDefaultFloat32Value(self):
        structureDefaultValues = self.api.StructureDefaultValues()
        self.assertEqual(1.234, structureDefaultValues.float32_value)

    def testDefaultFloat64Value(self):
        structureDefaultValues = self.api.StructureDefaultValues()
        self.assertEqual(1.2345, structureDefaultValues.float64_value)

    def testDefaultStringValue(self):
        structureDefaultValues = self.api.StructureDefaultValues()
        self.assertEqual("string", structureDefaultValues.string_value)

    def testDefaultEnumValue(self):
        structureDefaultValues = self.api.StructureDefaultValues()
        self.assertEqual(self.api.BasicColor.BLACK, structureDefaultValues.enum_value)

    def testDefaultBitmaskValue(self):
        structureDefaultValues = self.api.StructureDefaultValues()
        self.assertEqual(self.api.Permission.Values.READ_WRITE, structureDefaultValues.bitmask_value)

File: test/extensions/language/default_values/cpp/StructureDefaultValuesTest.cpp
#include "default_values/structure_default_values/BasicColor.h"
#include "default_values/structure_default_values/Permission.h"
#include "default_values/structure_default_values/StructureDefaultValues.h"
#include "gtest/gtest.h"
#include "zserio/BitStreamReader.h"
#include "zserio/CppRuntimeException.h"

namespace default_values
{
namespace structure_default_values
{

TEST(StructureDefaultValuesTest, checkDefaultBoolValue)
{
    StructureDefaultValues structureDefaultValues;
    ASSERT_EQ(true, structureDefaultValues.getBoolValue());
}

TEST(StructureDefaultValuesTest, checkDefaultBit4Value)
{
    StructureDefaultValues structureDefaultValues;
    ASSERT_EQ(0x0F, structureDefaultValues.getBit4Value());
}

TEST(StructureDefaultValuesTest, checkDefaultInt16Value)
{
    StructureDefaultValues structureDefaultValues;
    ASSERT_EQ(0x0BEE, structureDefaultValues.getInt16Value());
}

TEST(StructureDefaultValuesTest, checkDefaultFloat16Value)
{
    StructureDefaultValues structureDefaultValues;
    float diff = 1.23F - structureDefaultValues.getFloat16Value();
    if (diff < 0.0F)
    {
        diff = -diff;
    }
    ASSERT_TRUE(diff <= std::numeric_limits<float>::epsilon());
}

TEST(StructureDefaultValuesTest, checkDefaultFloat32Value)
{
    StructureDefaultValues structureDefaultValues;
    float diff = 1.234F - structureDefaultValues.getFloat32Value();
    if (diff < 0.0F)
    {
        diff = -diff;
    }
    ASSERT_TRUE(diff <= std::numeric_limits<float>::epsilon());
}

TEST(StructureDefaultValuesTest, checkDefaultFloat64Value)
{
    StructureDefaultValues structureDefaultValues;
    double diff = 1.2345 - structureDefaultValues.getFloat64Value();
    if (diff < 0.0)
    {
        diff = -diff;
    }
    ASSERT_TRUE(diff <= std::numeric_limits<double>::epsilon());
}

TEST(StructureDefaultValuesTest, checkDefaultStringValue)
{
    StructureDefaultValues structureDefaultValues;
    ASSERT_EQ("string", structureDefaultValues.getStringValue());
}

TEST(StructureDefaultValuesTest, checkDefaultEnumValue)
{
    StructureDefaultValues structureDefaultValues;
    ASSERT_EQ(BasicColor::BLACK, structureDefaultValues.getEnumValue());
}

TEST(StructureDefaultValuesTest, checkDefaultBitmaskValue)
{
    StructureDefaultValues structureDefaultValues;
    ASSERT_EQ(Permission::Values::READ_WRITE, structureDefaultValues.getBitmaskValue());
}

} // namespace structure_default_values
} // namespace default_values
